### PR TITLE
Move mypy config from setup.cfg to pyproject.toml

### PR DESCRIPTION
**Pull request by jcwchen**
### Description
Move mypy config from setup.cfg to pyproject.toml

### Motivation and Context
Closes https://github.com/onnx/onnx/issues/4538.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,3 +1,39 @@
...
+no_implicit_optional = true
+# TODO disallow_incomplete_defs = true
+# TODO disallow_subclassing_any = true
+disallow_untyped_decorators = true
+warn_unused_configs = true
+show_error_codes = true
+show_column_numbers = true
+ignore_missing_imports = true
+exclude = [
+    '\.git',
```
Comment (justinchuby) [inline pyproject.toml:20]: We can use vscode's format document to standardize format on this file

**Review by justinchuby — APPROVED** [review]
LGTM🚀

File: setup.cfg [hunk 2]
```diff
@@ -42,49 +42,6 @@ exclude =
...
-
-[mypy-tools.*]
-disallow_untyped_defs = True
-
-# Ignore errors in setup.py
-[mypy-setup]
-ignore_errors = True
-
-
 [pylint.messages_control]
```
Comment (justinchuby) [inline setup.cfg:46]: fyi this can also go into pyproject I think

File: pyproject.toml (hunk 1 above)
Comment (jcwchen) [reply pyproject.toml:20]: Formatted and moved pylint. Thanks!

**Review by liqunfu — APPROVED** [review]

**Pull request merged**